### PR TITLE
[SYCL] Copy L0 loader to build dir if path to loader is provided

### PR DESCRIPTION
**Pull request by againull**
Signed-off-by: Artur Gainullin <artur.gainullin@intel.com>

### CODE REVIEW

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 1]
```diff
@@ -46,8 +46,17 @@ if (NOT DEFINED L0_LIBRARY OR NOT DEFINED L0_INCLUDE_DIR)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS l0-loader)
 else()
   include_directories("${L0_INCLUDE_DIR}")
-  add_custom_target(l0-loader
-    DEPENDS ${L0_LIBRARY} COMMENT "Using provided L0 Loader")
+  include_directories("${L0_INCLUDE_DIR}")
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:49]: Duplicate of line 48

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 2]
```diff
@@ -46,8 +46,17 @@ if (NOT DEFINED L0_LIBRARY OR NOT DEFINED L0_INCLUDE_DIR)
...
+  file(GLOB L0_LIBRARY_SRC "${L0_LIBRARY}*")
+  file(COPY ${L0_LIBRARY_SRC} DESTINATION ${LLVM_LIBRARY_OUTPUT_INTDIR})
+  add_custom_target(l0-loader DEPENDS ${L0_LIBRARY} COMMENT "Copying Level Zero Loader ...")
+  if(MSVC)
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:53]: lines 53-59 are the same as 5-11, can't you move it before line 3?

Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:53]: Please remind me why MSVC uses static library?

Comment (againull) [reply sycl/plugins/level_zero/CMakeLists.txt]: It is not a static library, it is an import library for .dll. To use the dll, application should link with this import library. It just has the same .lib extension as static libraries.

Comment (smaslov-intel) [reply sycl/plugins/level_zero/CMakeLists.txt]: OK, then I am surprised that `cmake` doesn't have something for it. Linking is always to .lib on Windows and .so on Linux. Even if there is a better way to write this, it shouldn't block this PR.

**Pull request merged**